wrappers/core18.go

```diff
@@ -202,19 +223,22 @@ func AddSnapdSnapServices(s *snap.Info, inter interacter) error {
 		// nothing to do
 		return nil
 	}
-	// stop all removed units first
-	for _, unit := range removed {
-		serviceUnits := []string{unit}
-		if err := sysd.Stop(serviceUnits, 5*time.Second); err != nil {
-			logger.Noticef("failed to stop %q: %v", unit, err)
-		}
-		if err := sysd.Disable(serviceUnits); err != nil {
-			logger.Noticef("failed to disable %q: %v", unit, err)
+
+	if !opts.Preseeding {
```

Comment:
There is a lot of code in this function that is disabled when preseeding; IMHO it would make sense to try to move that to a separate function, which is only invoked when not preseeding.

Reply:
Thanks, this might be a good idea indeed, I'll look into it.